### ISSUE TITLE
Server throws an InterfaceError if no one interacts with it after x number of hours

### ISSUE THREAD

**Issue by ecobost** [fabiansinz/pipecontrol#12]
This happens when the datajoint connection gets closed after some time of inactivity (72 hours). I think we already tried reconnecting with the datajoint server (so run dj.conn() before any server call) but it didn't work (not sure what where the effects). Datajoint does have the `dj.config['database.reconnect']` boolean flag so maybe this will solve it.

If it doesn't,  we could add a daemon that asks for something to the flask server every 12 hours or so. This could even run inside the docker container (started in the ./run_server.sh script) so it's all self contained. And it should send some signal to the server just so it calls dj.ping() and keeps the connection alive.
So in ./run_server.sh
```bash
watch -n 43200 curl shikigami.ad.bcm.edu/keepalive 
# this won't quite work, we should daemonize it and put it before the `flask run` call so it
# returns control to the main thread and then `flask run` is executed, `flask run` should
# be the last thing to execute so we can access the flask server outputs via docker logs
```
and in views.py
```python
@main.route('/keepalive')
def keepalive():
    dj.ping()
    return '', http.HTTPStatus.NO_CONTENT # https://www.erol.si/2018/03/flask-return-204-no-content-response/
```

**Comment by ecobost** [comment]
Found out datajoint version 0.11.1, latest version in pip has a bug in the reconnection but it is fixed in the development version (in Github). Also datajoint 0.12.0 (still unreleased) will have reconnect as True by default so this issue might be a non-issue. I'll run some tests with the development version of datajoint and report back. If it works, we'll just have to wait for release 0.12.0

**Comment by ecobost** [comment]
I've tested the reconnection with datajoint (database.reconnect=True). It does not quite work. When a user tries to access the db after 72 hours without any interaction, a warning is printed to the Flask console "Mysql server has gone away. Reconnecting to the server", connection successfully reconnects and it allows you to surf through the site and make changes like deleting a job from the jobs tables or adding a mouse in Autoprocessing. However, when I go to check in the actual MySQL tables, the changes are not there, for instance, the new entries will not be in AutoProcessing. Somehow it creates a parallel set of tables where it applies those changes but never commits them to the real tables. I've had this happened before both on shikigami and in a normal notebook so it may be a datajoint bug.

**Comment by ecobost** [comment]
Reconnection works but there is a bug with it https://github.com/datajoint/datajoint-python/issues/597 Once that's solved, this issue will be solved too

**Comment by ecobost** [comment]
Problem has being solved with https://github.com/datajoint/datajoint-python/pull/599 which will appear in dj 0.12.0. We'll wait until then to test this again.

**Comment by ecobost** [comment]
fixed in datajoint 0.11.3
